addons/packages/pinniped/0.4.4/bundle/config/overlay/update_overlay.yaml

```diff
@@ -0,0 +1,31 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+
+#@overlay/match by=overlay.subset({"kind":"DaemonSet"}), expects="0+"
+---
+spec:
+  #@overlay/match missing_ok=True
+  updateStrategy:
+    #@overlay/match missing_ok=True
+    type: OnDelete
+
+#@overlay/match by=overlay.subset({"kind":"Deployment"}), expects="0+"
+---
+spec:
+  #@overlay/match missing_ok=True
+  strategy:
+    #@overlay/match missing_ok=True
+    type: RollingUpdate
+    #@overlay/match missing_ok=True
+    rollingUpdate:
+      #@overlay/match missing_ok=True
+      maxUnavailable: 0
+      maxSurge: 1
+  template:
+    spec:
+    #@ if data.values.tanzuKubernetesRelease and data.values.tanzuKubernetesRelease != "":
+      #@overlay/match missing_ok=True
+      nodeSelector:
+        #@overlay/match missing_ok=True
+        tanzuKubernetesRelease: #@ data.values.tanzuKubernetesRelease
```

Comment:
i don't think that pinniped has a dependency on the kube version right now, i.e., if we update the package version and then we update the cluster version then the pinniped package will still continue to function (unlike CNI). are we sure it makes sense to add this logic to the package even though it is unnecessary? it seems like it just bloats the package with logic that we don't need.